### PR TITLE
fix: performance decline issue

### PR DESCRIPTION
**Pull request by sesidhar01**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

Bug fix

**What is the current behavior? (You can also link to an open issue here)**

https://github.com/ngx-formly/ngx-formly/issues/3984

**What is the new behavior (if this is a feature change)?**

**Please check if the PR fulfills these requirements**
- [x] The commit messages follow our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [x] A unit test has been written for this change.
- [x] Running `npm run build` produced a successful build. (Unit testing can be done by running `npm test`;)
- [x] My code has been linted. (`npm run lint` to do this. `npm run build` will fail if there are files not linted.)

**Please provide a screenshot of this feature before and after your code changes, if applicable.**

**Other information**:

### CODE REVIEW

File: src/core/src/lib/components/formly.field.ts [hunk 1]
```diff
@@ -283,12 +283,9 @@ export class FormlyField implements DoCheck, OnInit, OnChanges, AfterContentInit
...
       return () => {};
     }
-
-    const subscribes = [observeDeep(field, ['props'], () => field.options.detectChanges(field))];
-
-    if (field.options) {
-      subscribes.push(observeDeep(field.options, ['formState'], () => field.options.detectChanges(field)));
-    }
+    
+    const propsObserver = observe(field, ['props'], () => field.options.detectChanges(field));
```
Comment (aitboudad) [inline src/core/src/lib/components/formly.field.ts:287]: I think we should keep `observeDeep`. if there is a specific case where the perf drop we need to provide a fix for. `observeDeep` instead.

File: src/core/src/lib/components/formly.form.ts [hunk 2]
```diff
@@ -141,11 +142,16 @@ export class FormlyForm implements DoCheck, OnChanges, OnDestroy {
           // runGuarded is used to keep in sync the expression changes
           // https://github.com/ngx-formly/ngx-formly/issues/2095
           this.checkExpressionChange();
+          if(this.field.options){
+            fieldChangesDetection.push(() => observe(this.field.options, ['formState'], () => this.field.options.detectChanges(this.field)));
```
Comment (aitboudad) [inline src/core/src/lib/components/formly.form.ts:146]: use `observeDeep`

**Review by aitboudad — CHANGES_REQUESTED** [review]

**Review by aitboudad — APPROVED** [review]

**Comment by aitboudad** [comment]
Thank you @sesidhar01!

**Comment by kenisteward** [comment]
@sesidhar01  thanks buddy!  We'll get this into our wrapper lib ASAP

**Comment by sesidhar01** [comment]
> Thank you @sesidhar01!

@aitboudad Thanks for merging any idea when this change can be published?

**Comment by aitboudad** [comment]
> Thanks for merging any idea when this change can be published?

I'll try to make a new release at the end of this week.

**Pull request merged**